Update advanced_log_collection.md (#21816)
DataDog/documentation: content/en/agent/logs/advanced_log_collection.md

   - \d{4}\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01])
   - '[A-Za-z_]+ \d+, \d+ \d+:\d+:\d+ (AM|PM)'
```

> For containerized deployments, you can enable `auto_multi_line_extra_patterns` by passing a comma-separated string in the `DD_LOGS_CONFIG_AUTO_MULTI_LINE_EXTRA_PATTERNS` environment variable.

Reviewer: From a quick test, I think it needs to be in the form of a JSON array with proper escaping of quotes: 
`DD_LOGS_CONFIG_AUTO_MULTI_LINE_EXTRA_PATTERNS=[\"a\",\"b\"]`

Author: @gh123man as in, each value in the array needs to be a JSON? I'd set it as a the string array based on implementing it following [this](https://github.com/DataDog/logs-psa/blob/57e270c62df03338a98425970217ea0a1716ead9/POCs/containers/java/multi-line-logging/README.md?plain=1#L66) workaround method, and how the config appears to be interpretted in the logs source [code](https://github.com/DataDog/datadog-agent-private/blob/4cd0455fd8ace40070baaedd6b6bf28990b4fdd0/pkg/logs/internal/decoder/decoder.go#L172). I will test out your suggestion.

Reviewer: The way our config parser works, a single value `DD_LOGS_CONFIG_AUTO_MULTI_LINE_EXTRA_PATTERNS=foo` will still parse as a string slice `[]string{}` of size `1`. But in order to get more than one value, we have to format it as a JSON array. 

So each value in the JSON array needs to be a valid JSON string, and appropriately escaped for the shell (or config). The example linked above only uses 1 value so this is not very clear.

> For containerized deployments, you can enable `auto_multi_line_extra_patterns` with the `DD_LOGS_CONFIG_AUTO_MULTI_LINE_EXTRA_PATTERNS` environment variable, by passing it a comma separated string.

Reviewer 2: ```suggestion
For containerized deployments, you can enable `auto_multi_line_extra_patterns` by passing a comma-separated string in the `DD_LOGS_CONFIG_AUTO_MULTI_LINE_EXTRA_PATTERNS` environment variable.
```
